frontend/src/metabase-lib/types.ts

```diff
@@ -156,6 +156,13 @@ export type ExternalOp = {
   args: [ColumnMetadata, ...ExpressionArg[]];
 };
 
+export type FilterParts = {
+  operator: FilterOperator;
+  options: Record<string, unknown>;
+  column: ColumnWithOperators | null;
```

Comment:
`null` is returned if the reference in the first arg of the filter cannot be resolved to a column. This should not happen, but currently it does. When #32698 gets merged, `filterParts` will be able to resolve broken legacy references too.